### PR TITLE
[MM-60484 & MM-60485] Add disabled notification banners and section notices for web

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60484
https://mattermost.atlassian.net/browse/MM-60485

#### Screenshots

CleanShot 2024-10-09 at 17 59 29

CleanShot 2024-10-09 at 18 00 45

CleanShot 2024-10-09 at 18 01 22

CleanShot 2024-10-09 at 18 01 43

CleanShot 2024-10-09 at 18 01 54

#### Release Note

```release-note

```

### CODE REVIEW

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed. Still testing out all the different combinations, but starting a review for the issues I found. 
1. Collapsed state indicator - Can we match the colours to the design?
image

2. Not a major issue but it is weird that the notice goes away on simply closing the browser permission prompt. It does come back if you navigate away and come back. Is there any way to prevent this?

https://github.com/user-attachments/assets/ce0e29e2-a068-4531-b42b-7dfff3af9fac

Same happens if you deny the permissions. Ideally the notice should switch to the permission denied state instead of disappearing.
  
3. In the permission denied case, can we update the styling and text of the button to  match the design.
image

Also, it can probably be linked to [this docs page](https://docs.mattermost.com/preferences/manage-your-notifications.html) for now but later we'll link it to the missing notifications section that we're working on as a part of the `Troubleshooting notifications` documentation.

4. For the global banner, can we change the copy for web browser's to "We need your permission to show web browser notifications." instead of "We need your permission to show desktop notifications." as per [Carrie's suggestion in Figma](https://www.figma.com/design/Ct22FpS3qX5uAGdUFbN5qf?node-id=1318-10235#959563623).

5. Text in the `Enable notifications` button in the global banner is vertically misaligned. Changing line height of the text to 14px does the trick.
image

**Comment by abhijit-singh** [comment]
Continuing with the review...

6. In the desktop app, if I `deny` the permission in the initial modal, or if I hit `esc` to dismiss that modal and neither allow nor grant the permission, I don't see any global banner or an indication in the settings. You can notice in the `Edit Server` modal that permissions have not been granted hence the indication should be there in settings. 


https://github.com/user-attachments/assets/02230338-2edc-45ba-abcb-895597f3ca37

**Comment by M-ZubairAhmed** [comment]
> Continuing with the review but just looking at the title again, does this PR only implement the indicators on Web and not in the desktop app? You can ignore the below comment if it doesn't add them to the desktop app.
> 
>     6. In the desktop app, if I `deny` the permission in the initial modal, or if I hit `esc` to dismiss that modal and neither allow nor grant the permission, I don't see any global banner or an indication in the settings. You can notice in the `Edit Server` modal that permissions have not been granted hence the indication should be there in settings.
> 
> 
> Screen.Recording.2024-10-07.at.4.58.57.PM.mov

Yes this is just for web

**Comment by M-ZubairAhmed** [comment]
> Thanks @M-ZubairAhmed. Still testing out all the different combinations, but starting a review for the issues I found.
> 
>     1. Collapsed state indicator - Can we match the colours to the design?
>        image
> 
>     2. Not a major issue but it is weird that the notice goes away on simply closing the browser permission prompt. It does come back if you navigate away and come back. Is there any way to prevent this?
> 
> 
> Screen.Recording.2024-10-07.at.4.29.54.PM.mov
> 
> Same happens if you deny the permissions. Ideally the notice should switch to the permission denied state instead of disappearing.
> 
>     3. In the permission denied case, can we update the styling and text of the button to  match the design.
> 
> 

> 
> Also, it can probably be linked to [this docs page](https://docs.mattermost.com/preferences/manage-your-notifications.html) for now but later we'll link it to the missing notifications section that we're working on as a part of the `Troubleshooting notifications` documentation.
> 
>     4. For the global banner, can we change the copy for web browser's to "We need your permission to show web browser notifications." instead of "We need your permission to show desktop notifications." as per [Carrie's suggestion in Figma](https://www.figma.com/design/Ct22FpS3qX5uAGdUFbN5qf?node-id=1318-10235#959563623).
> 
>     5. Text in the `Enable notifications` button in the global banner is vertically misaligned. Changing line height of the text to 14px does the trick.
> 
> 

Fixed all the U.X reviews. @abhijit-singh can you re check if there are other issues. Please also check the copy text as well

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed! The changes look great and almost everything is working perfectly imo. Just a few last things:
1. In the state when notification permissions are denied, I don't think we need to mention 'web browser' in the button label. It can just be 'How to enable notifications' instead of 'How to enable web browser notifications'.
image

2. With regards to this issue https://mattermost.atlassian.net/browse/MM-60964 and the topic of adding the dismiss button, can we make the following changes:
 - Update the button in the permissions banner to “Manage Notification Preferences” instead of "Enable Notifications" cc @cwarnermm open to other suggestions here (more context in this [issue and the comments on it](https://mattermost.atlassian.net/browse/MM-60964))
 - If the user clicks on the 'X' icon to dismiss the banner, it should go away on that particular browser until cookies get cleared or the session expires. Currently it shows up again on a simple refresh.
 
Also open to if you want to do this in a separate PR. 

Note: I'll also test on other operating systems and browsers, which I'd recommend we also do while doing QA testing on this.

**Comment by M-ZubairAhmed** [comment]
@abhijit-singh 

> 1. In the state when notification permissions are denied, I don't think we need to mention 'web browser' in the button label. It can just be 'How to enable notifications' instead of 'How to enable web browser notifications'.

Changes done

> With regards to this issue https://mattermost.atlassian.net/browse/MM-60964 and the topic of adding the dismiss button, can we make the following changes:

Lets take it up in separate PR and not this one

**Review by abhijit-singh — APPROVED** [review]

File: webapp/channels/src/components/widgets/tag/tag.tsx
Comment (hmhealey) [inline webapp/channels/src/components/widgets/tag/tag.tsx:12]: @abhijit-singh I don't see this variant in https://compass.mattermost.com/29be2c109/p/46ae87-tag. Should that be included there?

File: webapp/channels/src/components/widgets/tag/tag.tsx
Comment (hmhealey) [inline webapp/channels/src/components/widgets/tag/tag.tsx:29]: Why change all of these? The existing version of this code looks a bit odd, but I don't think we should be changing it right now

File: webapp/channels/src/sass/routes/_settings.scss
Comment (hmhealey) [inline webapp/channels/src/sass/routes/_settings.scss:271]: Shouldn't this be a `display: block` when the width is 100%?

**Review by hmhealey — CHANGES_REQUESTED** [review]

**Review by pvev — APPROVED** [review]
LGTM! thanks @M-ZubairAhmed

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx
Comment (hmhealey) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx:4]: Those ARIA attributes aren't actually needed as long as it's using a proper button or anchor element (although we use the wrong element) and the element contains text which I think it does in this case.

As for the focus, I don't think that should be moved here automatically since I think the post textbox is still the more important thing that the user would want to be focused on, and moving the user's focus unexpectedly can cause accessibility issues. I just learned about [the status role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role) which is seems like the announcement bar should use, but that's outside the scope of this PR and I want to read up more on that to make sure we don't do it wrong

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx:15]: ah missed this one. Changed

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/index.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/index.tsx:32]: changed

File: webapp/channels/src/components/setting_item_max.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/setting_item_max.tsx:48]: No inputs have different structure that the above one.

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx:18]: i dont think there is any meaningful gain with adding useCallback here

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx:12]: done

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/index.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/index.tsx:18]: removed

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:16]: Done ✅

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_denied_section_notice.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_denied_section_notice.tsx:22]: changed

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_never_granted_section_notice.tsx:18]: ok

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_denied_section_notice.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/notification_permission_section_notice/notification_permission_denied_section_notice.tsx:26]: Only 2 are same, i think its fine having multiple copies

File: webapp/channels/src/components/widgets/tag/tag.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/widgets/tag/tag.tsx:12]: Yes dim version of all/some needs to be updated

File: webapp/channels/src/components/widgets/tag/tag.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/widgets/tag/tag.tsx:29]: Agreed its not related, but the end code looks a lot more readable.

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_unsupported_bar.tsx:16]: Thought about that, but browser shows a UI interaction when its blocked the popup, and app shows indicator to open. So visually that should be enough for user to see its a trying to open a link

**Review by yasserfaraazkhan — APPROVED** [review]
- Closing banner and reloading will make it reappear
- If Browser Notification has been Blocked. The banner wont show up. Section notice is visible.
- Clicking on Enable notification from Section Notice, the browser `Ask` to `allow` or `block`. Once we allow the section notice goes away.
- Disabling the Browser notification at OS level. We can still Enable notification from Global header or Section notice.

File: webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/index.test.tsx
Comment (hmhealey) [reply webapp/channels/src/components/user_settings/notifications/desktop_and_mobile_notification_setting/index.test.tsx:21]: I'm curious what you had fought with here, but there's no reason to block this PR on it

File: webapp/channels/src/components/widgets/tag/tag.tsx
Comment (abhijit-singh) [reply webapp/channels/src/components/widgets/tag/tag.tsx:12]: Yep, I have it on my list to add that variation to the docs.

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx
Comment (abhijit-singh) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:45]: Well I think `web browser notifications` technically makes sense but I also agree that almost all other products call them `desktop notifications`. I wouldn't want to change it to just "We need your permission to show notifications" since it can leave room for ambiguity and could be confused with in-product mention badges and what not. 

I'm open to reverting back to `desktop notifications` but I'd want @cwarnermm to share thoughts and take the final call on this since this change was recommended by her.

Comment (cwarnermm) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:45]: How about "We need your permission to show notifications in this web browser" ?

Comment (abhijit-singh) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:45]: I don't mind it. I was thinking it's getting a bit long but we have the space and it doesn't truncate until the screen size is less than ~805px so that's okay.

image

Thoughts @hmhealey?

Comment (hmhealey) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:45]: How about "We need your permission to show notifications in the browser"? I still have a vendetta against calling it a "web browser" apparently, and while "this" technically correct that it only refers to the one browser, I feel like that doesn't matter to most people since they just use a single one

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx]: changed as discussed above

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:50]: I think this is fine because the context here has been changed

File: webapp/channels/src/sass/routes/_settings.scss
Comment (M-ZubairAhmed) [reply webapp/channels/src/sass/routes/_settings.scss]: agreed, i changed it

File: webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx
Comment (hmhealey) [reply webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx:50]: I think the change of context only matters if the text actually changes, but I'm fine leaving it since I've definitely blocked this PR for long enough with my other text changes 😅

**Review by hmhealey — APPROVED** [review]
Thanks with putting up with my nitpicking over wording. This looks good to me!

**Pull request merged**
